### DISCUSSION TITLE
User text input on a an Edge

### DISCUSSION THREAD

**Discussion by sunilrav** [xyflow/xyflow#3377] [General]
Looking for help or an example of where a user can input text on an edge and the app can keep track of it between save and restore.

I have the below code, but I don't know how to persist it across save and restore.

`

<BaseEdge path={edgePath} markerEnd={markerEnd} style={style} />

      <EdgeLabelRenderer>

        <div

          style={{
            position: 'absolute',
            transform: `translate(-50%, -50%) translate(${labelX}px,${labelY}px)`,
            fontSize: 12,
            // everything inside EdgeLabelRenderer has no pointer events by default
            // if you have an interactive element, set pointer-events: all
            pointerEvents: 'all',
          }}
          className="nodrag nopan"
        >
     
         <input id='edgetext' className='edgelabel nodrag' type='text' onChange={onChange} />
          
        </div>
      </EdgeLabelRenderer>
`

**Comment by p-shepherd** [comment]
I have done something like this

onConnect create a new connection with a new edge

const newEdge = {
        id: randomEdge,
        ...connection,
        style: strokeThings,
        type: strokeType,
        label: inputValue2,
        labelBgPadding: [8, 4],
        labelBgBorderRadius: 4,
        labelBgStyle: { fill: "orange", color: "black", fillOpacity: 0.9 }
      }; // Add style property with desired color
      
      label value will be based on user input, if none then it will not appear
      
      this is in main app.js file
      
      newEdge is then passed to edges, setEdges state variable
